woocommerce/woocommerce
# Add suggested-products product endpoint

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





This PR introduces a new Suggested Products endpoint:

```
wc/v2/products/suggested-products
```

### Params

The endpoint accepts three parameters: `categories`, `tags`, and `limit`.

```
wc/v2/products/suggested-products?related_categories[]=<cat-01>&related_categories[]=<cat-02>&tags[]=<tag-01>...
```

Also, it accepts the `combine` param to combine or not the current product items (related_categories, related_tags, ...) with the provided ones.

Closes https://github.com/woocommerce/woocommerce/issues/43714



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

A simple way to check this endpoint if by using the apiFetch library which is exposed globally in the new Product Editor app:

1. Enable the new Product Editor app
2. Create/edit a product
3. Open the dev console

Now, you can start to perform requests by using the apiFetch lib:

```
wp.apiFetch( { path: 'wc/v3/products/suggested-products' } ).then( console.log ).catch( console.error )
```

### Get default suggested products

1. Ensure you have a few products to test.
2. Set categories and tags to get suggested products
3. Get the suggested products by running the following script. Remember you need to define the parameters for your testing site. I've defined category `21`, and limit `3`

```js
wp.apiFetch( { path: 'wc/v3/products/suggested-products?categories=21&limit=3' } )
    .then( console.log )
    .catch( console.error )
```

<img width="853" alt="image" src="https://github.com/woocommerce/woocommerce/assets/77539/61879b8a-6328-4847-8d01-b0fbd228d6df">



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
